File: modules/agent/src/models/Signal.py
"""Signal model for intent signals on Accounts."""

from sqlalchemy import Column, Text, DateTime, BigInteger, Date, Numeric, func, ForeignKey
from sqlalchemy.orm import relationship
from models import Base


class Signal(Base):
    """Signal SQLAlchemy model for intent signals."""

    __tablename__ = "Signal"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    account_id = Column(BigInteger, ForeignKey("Account.id", ondelete="CASCADE"), nullable=False)
    signal_type = Column(Text, nullable=False)   # 'hiring', 'expansion', 'product_launch', 'partnership', 'leadership_change', 'news'
    headline = Column(Text, nullable=False)
    description = Column(Text, nullable=True)
    signal_date = Column(Date, nullable=True)
    source = Column(Text, nullable=True)         # 'linkedin', 'news', 'crunchbase', 'agent'
    source_url = Column(Text, nullable=True)
    sentiment = Column(Text, nullable=True)      # 'positive', 'neutral', 'negative'
    relevance_score = Column(Numeric(3, 2), nullable=True)  # 0.00 to 1.00
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    # Relationships
    account = relationship("Account", back_populates="signals")

File: modules/agent/internal/models/Signal.go
package models

import (
	"time"

	"github.com/shopspring/decimal"
)

// Signal represents an intent signal on an Account
type Signal struct {
	ID             int64            `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	AccountID      int64            `gorm:"column:account_id;not null" json:"account_id"`
	SignalType     string           `gorm:"column:signal_type;not null" json:"signal_type"` // 'hiring', 'expansion', 'product_launch', 'partnership', 'leadership_change', 'news'
	Headline       string           `gorm:"column:headline;not null" json:"headline"`
	Description    *string          `gorm:"column:description" json:"description"`
	SignalDate     *time.Time       `gorm:"column:signal_date" json:"signal_date"`
	Source         *string          `gorm:"column:source" json:"source"`     // 'linkedin', 'news', 'crunchbase', 'agent'
	SourceURL      *string          `gorm:"column:source_url" json:"source_url"`
	Sentiment      *string          `gorm:"column:sentiment" json:"sentiment"` // 'positive', 'neutral', 'negative'
	RelevanceScore *decimal.Decimal `gorm:"column:relevance_score;type:numeric(3,2)" json:"relevance_score"`
	CreatedAt      time.Time        `gorm:"column:created_at;not null;default:now()" json:"created_at"`
	UpdatedAt      time.Time        `gorm:"column:updated_at;not null;default:now()" json:"updated_at"`

	// Relationships
	Account *Account `gorm:"foreignKey:AccountID" json:"account,omitempty"`
}

// TableName specifies the table name for Signal
func (Signal) TableName() string {
	return "Signal"
}
